### PR TITLE
Add support for H.264 profiles besides Constrained Baseline Profile

### PR DESCRIPTION
**Pull request by hensmi-amazon**
**Issue #:** None

**Description of changes:** Additional profiles are desired as they are often more likely to use hardware encoder then H.264 CBP. This change also includes changes that allows 'falling-forwards' to more preferred send codecs if limited receivers leave the call.

This also has a fix for the protobuf generation script since node@10 is not available on `brew` anymore.

**Testing:**

*Can these tested using a demo application? Please provide reproducible step-by-step instructions.*
1. Select H.264 Baseline, Main, or High from Additional options before joining the meeting on Chrome or Safari.
2. Enable video. Join from another Chrome device.
3. Join a third attendee from Firefox that will not have support for these profiles. The sender should fall back.
4. Leave the Firefox attendee. Note that there is a known issue with H.264 High profile on MacOS and there is a bug open with WebRTC, so this will lead to freezes currently. https://bugs.chromium.org/p/chromium/issues/detail?id=1520287 . The docstring indicates that this should not be used with fallbacks.

**Checklist:**

1. Have you successfully run `npm run build:release` locally?
y

2. Do you add, modify, or delete public API definitions? If yes, has that been reviewed and approved?
yes but the APIs match existing patterns so review is not necessary.

3. Do you change the wire protocol, e.g. the request method? If yes, has that been reviewed and approved?
n

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: demos/browser/app/meetingV2/meetingV2.html [hunk 1]
```diff
@@ -180,6 +183,9 @@ <h5 class="modal-title" id="additional-options-modal-label">Additional Options</
             <option value="default">Meeting Default</option>
             <option value="vp8">VP8</option>
             <option value="h264ConstrainedBaselineProfile">H.264 Constrained Baseline Profile</option>
+            <option value="h264BaselineProfile">H.264 Baseline Profile</option>
+            <option value="h264MainProfile">H.264 Main Profile</option>
+            <option value="h264HighProfile">H.264 High Profile</option>
```
Comment (shi-su) [inline demos/browser/app/meetingV2/meetingV2.html:188]: So I guess we don't bother putting constrained high in as a demo option

**Review by shi-su — APPROVED** [review]

File: src/transceivercontroller/DefaultTransceiverController.ts [hunk 2]
```diff
@@ -435,7 +435,7 @@ export default class DefaultTransceiverController
     }
 
     if (track) {
-      transceiver.direction = 'sendrecv';
+      transceiver.direction = 'sendonly';
```
[the remark this answers is not in the document]
Comment (hensmi-amazon) [reply src/transceivercontroller/DefaultTransceiverController.ts:438]: Oh yikes good catch, this was just for testing.

File: demos/browser/app/meetingV2/meetingV2.html (hunk 1 above)
Comment (hensmi-amazon) [reply demos/browser/app/meetingV2/meetingV2.html:188]: Actually, I'm looking at an iOS safari SDP now and it seems to include only constrained high, and not high. I think i'll look more into the fallback logic we were discussing.

**Review by shi-su — DISMISSED** [review]

File: src/task/ReceiveVideoStreamIndexTask.ts [hunk 3]
```diff
@@ -220,25 +220,23 @@ export default class ReceiveVideoStreamIndexTask
...
+    }
+
+    if (newMeetingSupportedVideoSendCodecPreferences.length > 0) {
+      // We need to renegotiate if we are currently sending a codec that is no longer supported in the call, or if we can
+      // start sending a higher preference codec again (due to a constrained remote attendee leaving)
       if (
         this.context.currentVideoSendCodec !== undefined &&
-        !codecSupported &&
-        capability.equals(this.context.currentVideoSendCodec)
+        !newMeetingSupportedVideoSendCodecPreferences[0].equals(this.context.currentVideoSendCodec)
```
Comment (ltrung) [inline src/task/ReceiveVideoStreamIndexTask.ts:236]: So the first codec in videoSendCodecPreferences is always the current send codec?

File: src/task/AttachMediaInputTask.ts [hunk 4]
```diff
@@ -18,11 +18,7 @@ export default class AttachMediaInputTask extends BaseTask {
   async run(): Promise<void> {
     const transceiverController = this.context.transceiverController;
     transceiverController.setPeer(this.context.peer);
-    try {
-      transceiverController.setupLocalTransceivers();
-    } catch (error) {
-      throw error;
-    }
+    transceiverController.setupLocalTransceivers();
```
Comment (ltrung) [inline src/task/AttachMediaInputTask.ts:21]: so this would not throw any error ?

**Review by ltrung — APPROVED** [review]

File: demos/browser/app/meetingV2/meetingV2.html (hunk 1 above)
Comment (hensmi-amazon) [reply demos/browser/app/meetingV2/meetingV2.html:188]: Made change on backend to forward CH as H, and included Constrain High when 'High' option is chosen.

File: src/task/AttachMediaInputTask.ts (hunk 4 above)
Comment (hensmi-amazon) [reply src/task/AttachMediaInputTask.ts:21]: The catch just rethrows it :P. My linter was complaining about it so figured i'd clean it up.

File: src/task/ReceiveVideoStreamIndexTask.ts (hunk 3 above)
Comment (hensmi-amazon) [reply src/task/ReceiveVideoStreamIndexTask.ts:236]: I thought so...but now that I think about it there is a pretty unlikely edge case where the preference list starts with a codec that can be received by all clients (including the sender) but not sent. While this is not possible on any existing browser/codec, let me see if I can harden this.

File: src/task/AttachMediaInputTask.ts (hunk 4 above)
Comment (ltrung) [reply src/task/AttachMediaInputTask.ts:21]: I am blind :))

**Review by ltrung — DISMISSED** [review]

**Review by ltrung — APPROVED** [review]

**Pull request merged**